src/main/kotlin/com/example/Employee/services/BankAccountService.kt

```diff
@@ -76,7 +76,7 @@ public class BankAccountService(private val employeeRepository: EmployeeReposito
         if(!employeeRepository.existsById(employeeId)){
             throw EntityNotFoundException("employee not found")
         }
-        if(!bankAccountRepository.existsById(id)){
```

Comment:
Here as I used existsById, it was only checking bankAccount id and was returning true even if we give other employees id. 
For example:
Let us consider two employees with employeeId's as 5 and 6.
Employee 5 had bank accounts with id's 1 and 2.

If we send the following delete request:
http://localhost:8075/employees/6/bank_accounts/1.
This request should not delete the bank account with id 1 as it does not belong to employee 6.
But this request has deleted the bank account with id 1 as I used existsById only.

So, I changed it to existsByEmployeeEmployeeIdAndId, so that it will return false when the given employee doesn’t have a bank account with given id.

I found the same mistake in getAccount and updateAccount methods. Made the required changes in the "Fixed bugs in bankAccount Service" commit. Please check.

Reply:
Ok, makes sense.